components/collective-page/sections/About.js

```diff
@@ -52,7 +53,7 @@ const SectionAbout = ({ collective, canEdit, intl }) => {
           formatBeforeSubmit={v => (isEmptyValue(v) ? null : v)}
           prepareVariables={(collective, longDescription) => ({
             id: collective.id,
-            longDescription: isEmptyValue(longDescription) ? null : longDescription,
+            longDescription: isEmptyValue(longDescription) ? null : replaceImgPreviews(longDescription),
```

Comment:
I don't like that we're editing the content received by the API here, because:
- Any mistake in `replaceImgPreviews` will create an XSS injection on the most visited page of our website.
- Adding this helper every time we need to display the long description would be a pain, and what about people consuming our public API?
- We'll get inconsistent storing of the video if we merge this as is (old ones will have an iframe, new ones will have an img).
- The problem described in the abovementioned issue impacts the rich text editor, therefore the fix should be in the rich text editor - not in `sections/About`.

The behavior I would consider the safest is:
- Keep storing iframes in `longDescription` like it's done today
- In `RichTextEditor`, replace all iframes with images when loading the content
- Export a helper function to transform YouTube images back to iframes when saving the content

Why is it safer:
- We're still relying on API for sanitization, without editing the result. As long as `sanitize-html` is safe, we're safe.
- If someone finds a way to inject code in the process that transforms the iframe to img, only the rich text editor will get impacted, so you'll only be able to attack the other admins of your collective. That greatly limits the scope of the attack.

Reply:
@SudharakaP `The RichTextEditor doesn't have any [event](https://github.com/basecamp/trix#observing-editor-changes) to catch when saving the content` Trix won't provide anything for this because we trigger the save function from our own button. The value is saved from https://github.com/opencollective/opencollective-frontend/blob/main/components/collective-page/sections/About.js#L55

Reply:
Thanks. Yes, so I am trying to make sure we reduce the attack vector to the `RichTextEditor` hence as you suggested above, 

>The problem described in the above mentioned issue impacts the rich text editor, therefore the fix should be in the rich text editor - not in `sections/About`.

to get this transformation logic within the `RichTextEditor`. But given that we trigger the save function from out own button, I think we need to have this transformation from the images to iframes outside of the `RichTextEditor`. Does that make sense? 🤔

Reply:
Yep, I did not choose the right words above. My point was that the fix should be in how we process `RichTextEditor`'s data, not in the way we display the `longDescription`.

Ideally, we want to export a shared helper from `RichTextEditor` to handle that.